src/lib/support/ScopedBuffer.h

```diff
@@ -148,6 +151,11 @@ class ScopedMemoryBuffer : public Impl::ScopedMemoryBufferBase<MemoryManagement>
     const T * Get() const { return static_cast<const T *>(Base::Ptr()); }
     const T & operator[](size_t index) const { return Get()[index]; }
 
+    inline T * operator->() { return Get(); }
+    inline const T * operator->() const { return Get(); }
+    inline const T & operator*() const { return *Get(); }
+    inline T & operator*() { return *Get(); }
```

Comment:
I am not sure these make sense for ScopedMemoryBuffer.  Why is the first element of this array special?